content/articles/pointers-in-c#/index.md

```diff
@@ -0,0 +1,330 @@
+
+```c#
+using System;
+namespace UnsafeCodeApplication
+
+{
+    class Demo
+    {
+        public unsafe static void Main()
+        {
+            int[] array = { 5, 6, 7, 8, 9 };    
+            fixed (int* ptr = array)                
+            for (int k=0; k< 5; k++)
+            {
+                Console.WriteLine("Value of array[{0}]={1}", k, *(ptr + k));
+                Console.WriteLine("Address of array[{0}]={1}", k, (int)(ptr + k));
+            }
+            Console.ReadKey();
+        }
+
+    }
+
+}
+```
+
+And the output is:
+
+```
+Value of array[0]=5
+Address of the array[0]=-773935792
+Value of array[1]=6
+Address of the array[1]=-773935788
+Value of array[2]=7
+Address of the array[2]=-773935784
+Value of array[3]=8
+Address of the array[3]=-773935780
+Value of array[4]=9
+Address of the array[4]=-773935776
+```
+
+In the example above, we used a command to restrict the objects in the array to a fixed memory allocation. `fixed (int* ptr = array)` performs that request.
+
+### Pointers and methods
+In C#, the pointers can be passed as a contention to a strategy as the following:
+
+```c#
+using System;
+namespace UnsafeCodeApplication
+{
+    class Demo
+    {
+        public unsafe void Method()
+        {
+            int x = 60;
+            int y = 30;
+            int* ptr1 = &x;
+            int* ptr2 = &y;
+            Console.WriteLine(*ptr1);       
+            Console.WriteLine(*ptr2);       
+            Console.WriteLine((int)ptr1);   
+            Console.WriteLine((int)ptr2);   
+        }
+    }
+    class Example
+    {
+        public static void Main()
+        {
+            Demo d = new Demo();
+            d.Method();
+        }
+    }   
+}
+```
+
+The output is:
+
+```
+60
+30
+1748493636
+1748493632
+```
+
+Unmanaged codes are used with the method which has two variables `x` and `y` with values 50 and 20, respectively. Pointers `*ptr1` and `*ptr2` point to their memory addresses.
+
+### Conversions and pointers
+Pointer types do not inherit from objects in C#, and there is no way to convert pointer types to objects. As a result, pointers do not help boxing and unboxing. Conversions between different pointer types, as well as pointer types and integral types, are supported in C#.
+
+C# upholds explicit and implicit pointers changes inside unmanaged settings. In implicit pointers, conversions are from null type to pointer type and also from any pointer type to type void * type.
+
+For explicit pointers, the cast operator (()) is essential. The conversion is from to in this instance.
+
+1. The type of pointer to other forms of the pointer.
+2. To other pointer types: byte, sbyte, short, ushort, int, uint, long, ulong.
+3. Pointer type to sbyte, byte, uint, int, long, ulong, short, ushort types.
+
+An example of pointer conversion is illustrated below:
+
+```c#
+char k = 'U';  
+char *pk = &k;  
+void *px = pk; 
+int *pj = (int *) px; 
+```
```

Comment:
Exaplain this example. What conversions are happening here? (From what to what?)